tools/sysctlfix/main.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+)
+
+// This tool attempts to write a sysctl config file to the sysctl config directory with the highest precedence so
+// we can overwrite any other config and ensure correct sysctl options for Cilium to function.
+
+const (
+	sysctlD = "/etc/sysctl.d/"
+	// The 99-zzz prefix ensures our config file gets precedence over most if not all other files.
+	ciliumOverwrites = "99-zzz-override_cilium.conf"
+)
+
+var sysctlConfig = strings.Join([]string{
+	"# Disable rp_filter on Cilium interfaces since it may cause mangled packets to be dropped",
+	"net.ipv4.conf.lxc*.rp_filter = 0",
```

Comment:
I overlooked that we are completely disabling the filter. What about setting it to the `Loose` mode? That's all we should need to pass the packets with asymmetric routing. Details - https://github.com/cilium/cilium/issues/13130. That's the value we are currently setting as well.

Reply:
I set it to disabled since that seemed to be the suggested fix in #10645. According too @aanm, [this](https://github.com/systemd/systemd/commit/230450d4e4f1f5fc9fa4295ed9185eea5b6ea16e) commit in SystemD is the source of the issue. In the commit the `rp_filter` is changed from 1(Strict) to 2(Loose). In #10645 [this](https://github.com/systemd/systemd/commit/5d4fc0e665a3639f92ac880896c56f9533441307#diff-7816eed8ca6324f23a690cc5f58e6bf7) commit is mentioned as the culprit (also a change to 2(Loose))

The description of 'Loose' mode is:
> Loose mode as defined in RFC3704 Loose Reverse Path
        Each incoming packet's source address is also tested against the FIB
        and if the source address is not reachable via any interface
        the packet check will fail.
        
I am not sure if there are common use cases in Cilium where we handle packets for source addresses which don't appear in the FIB, but it wouldn't suppose me if there are.

Reply:
Cilium isn't the only entity that could be running on a host. The issues you've linked above indicate that systemd overwrites `rp_filter` settings set by Cilium. Regardless, I would like for us to understand why we would need to disable source address checks altogether in case of Cilium.